lib/Config.ts

```diff
@@ -0,0 +1,111 @@
+import os from 'os';
+import path from 'path';
+import fs from 'fs';
+import toml from 'toml';
+import { deepMerge } from './Utils';
+
+class Config {
+  public walliDir: string;
+  public btcdDir: string;
+  public logfile: string;
+  public loglevel: string;
+  public rpc: {
+    port: number,
+    host: string,
+    user: string,
+    password: string,
+  };
+
+  constructor() {
+    const platform = os.platform();
+
+    switch (platform){
+      // TODO: Check if the btcd.conf file directory is proper for linux and mac
+      case 'linux': {
+        const localDir = process.env.HOME;
+        this.walliDir = `${localDir}/.walli/`;
+        this.btcdDir = `${localDir}/Btcd/`;
+        break;
+      }
+      case 'darwin': {
+        const localDir = process.env.HOME;
+        this.walliDir = `${localDir}/.walli/`;
+        this.btcdDir = `${localDir}/Btcd/`;
+        break;
+      }
+      default: /* WIN32 */ {
```

Comment:
This is a minor detail but FreeBSD and OpenBSD would be treated like a Windows operating system. Use `win32` to detect Windows like systems and use `default` for Linux and the BSD ones.